src/librbd/AioImageRequest.cc

```diff
@@ -462,8 +462,12 @@ void AioImageFlush::send_request() {
     m_aio_comp->associate_journal_event(journal_tid);
   }
 
-  C_AioRequest *req_comp = new C_AioRequest(cct, m_aio_comp);
-  m_image_ctx.flush(req_comp);
+  if (!journaling && m_image_ctx.object_cacher != NULL) {
```

Comment:
I think this just becomes an "} else {" clause to the previous "if (journaling) {" conditional.